titanium/app/controllers/search.js

```diff
@@ -0,0 +1,186 @@
+// Create/Reference instances of all collections we need
+var properties = Alloy.Collections.instance('property');
+var favorites = Alloy.Collections.instance('favorite');
+var searches = Alloy.Collections.instance('search');
+
+// Self-executing constructor, just for clarity
+(function constructor(args) {
+
+  // Fetch favorites from SQL data source so we can use them in different controllers
+  favorites.fetch();
+
+  // Show/Hide the ambigious/recent locatins when collection updates and do so now
+  searches.on('fetch reset', showOptions);
```

Comment:
Only in this occasion it was missing and I just corrected it. It wouldn't cause any leaks in this case because the search window is never closed, but it's best practice to do so. Thanks!